src/test/java/io/nats/client/impl/RequestTests.java

```diff
@@ -239,6 +239,39 @@ public void testRequireCleanupOnTimeoutNoNoResponders() throws IOException, Inte
         }
     }
 
+    @Test
+    public void testRequireCleanupOnTimeoutCleanCompletable() throws IOException, InterruptedException {
```

Comment:
I'd like to see some tests that also succeed, and then fail with a timeout.  This one does test failure w/ cleanup - but would like to see that the right exception is thrown.

Reply:
Hi! For a "succeed and then fail" what do you mean exactly? I see that the timeout of the `NatsRequestCompletableFuture` is considered only upon cleanup.

I can add an additional test that triggers the timeout exception, coming from the `CompletableFuture`.
Something like:
`assertThrows(TimeoutException.class, () -> nc.requestWithTimeout(nm, Duration.ofMillis(cleanupInterval)).get(100, TimeUnit.MILLISECONDS));`

Thanks for your reply

Reply:
so this this is request/reply, the timeout is suggesting how long to wait for a response. Test actually getting a response back and also not getting a response back. Although No responders may come into play. What do other tests for the default timeout do?

Reply:
added more tests and api update